x-pack/plugins/apm/public/components/alerting/service_overview_alerts/alerts_table_status_filter.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonGroup, EuiButtonGroupOptionProps } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import {
+  ALERT_STATUS_ACTIVE,
+  ALERT_STATUS_RECOVERED,
+  ALERT_STATUS,
+} from '@kbn/rule-data-utils';
+export const ALL_ALERTS_FILTER = 'ALL_ALERTS_FILTER';
+
+export type AlertStatusFilterButton =
+  | typeof ALERT_STATUS_ACTIVE
+  | typeof ALERT_STATUS_RECOVERED
+  | typeof ALL_ALERTS_FILTER;
+
+export interface AlertStatusFilter {
+  status: AlertStatusFilterButton;
+  query: string;
+  label: string;
+}
+
+export interface AlertStatusFilterProps {
+  status: AlertStatusFilterButton;
+  onChange: (id: string, value: string) => void;
+}
+
+export const allAlerts: AlertStatusFilter = {
+  status: ALL_ALERTS_FILTER,
+  query: '',
+  label: i18n.translate('xpack.apm.alerts.alertStatusFilter.showAll', {
+    defaultMessage: 'Show all',
+  }),
+};
+
+export const activeAlerts: AlertStatusFilter = {
+  status: ALERT_STATUS_ACTIVE,
+  query: `${ALERT_STATUS}: "${ALERT_STATUS_ACTIVE}"`,
+  label: i18n.translate('xpack.apm.alerts.alertStatusFilter.active', {
+    defaultMessage: 'Active',
+  }),
+};
+
+export const recoveredAlerts: AlertStatusFilter = {
+  status: ALERT_STATUS_RECOVERED,
+  query: `${ALERT_STATUS}: "${ALERT_STATUS_RECOVERED}"`,
+  label: i18n.translate('xpack.apm.alerts.alertStatusFilter.recovered', {
+    defaultMessage: 'Recovered',
+  }),
+};
+
+const options: EuiButtonGroupOptionProps[] = [
+  {
+    id: allAlerts.status,
```

Comment:
You could inline all options in this array, if you prefer to keep it like this some variables / types above don't have to be exported.

Reply:
I just removed the export for the variables. I tend to think is more readable if it's separate and not inlined but I'm happy to change it if anyone feels different.